### PR TITLE
Backport random.choices from Python 3.6

### PR DESCRIPTION
**Pull request by kaapstorm**
Picks up from discussions on #23323 and #23336

### CODE REVIEW

File: corehq/sql_db/util.py [hunk 1]
```diff
@@ -290,7 +324,5 @@ def select_db_for_read(weighted_dbs):
             weights.append(weight)
 
     if dbs:
-        # normalize weights of remaining dbs
-        total_weight = sum(weights)
-        normalized_weights = [float(weight) / total_weight for weight in weights]
-        return random.choice(dbs, p=normalized_weights)
+        choices_func = random.choices if sys.version_info >= (3, 6) else choices_backport
```
Comment (emord) [inline corehq/sql_db/util.py]: MIght be nicer to have something like:

```python
try:
   from random import choices
except ImportError:
   def choices():
        ....
```

could also be extracted into another file.

**Review by emord — APPROVED** [review]
Any reason to not include your tests from the other PR? Probably not necessary as long as the function that uses this backport is well tested

File: corehq/sql_db/util.py (hunk 1 above)
Comment (dannyroberts) [reply corehq/sql_db/util.py]: +1

**Review by dannyroberts — APPROVED** [review]
Thanks Norman, this looks great!

File: corehq/sql_db/util.py (hunk 1 above)
Comment (kaapstorm) [reply corehq/sql_db/util.py]: Excellent idea.

**Comment by kaapstorm** [comment]
Moved to a new file to clean up the import. Re-added and updated tests, now with `random.seed()`.

**Review by emord — APPROVED** [review]

**Comment by emord** [comment]
Tests failing because this is dependent on removing MVP PR

**Comment by dannyroberts** [comment]
Sweet, took the liberty of merging in master to hopefully get the tests passing now that https://github.com/dimagi/commcare-hq/pull/23323 is merged (!)

**Review by millerdev — COMMENTED** [review]
Excellent!

**Pull request merged**